services/wallet/datastore.go

```diff
@@ -238,6 +240,20 @@ func (pg *Postgres) GetWalletByPublicKey(ctx context.Context, pk string) (*walle
 	return &wallet, err
 }
 
+// HasPriorLinking - check if this wallet id has been linked to this provider linking id in the past
+func (pg *Postgres) HasPriorLinking(ctx context.Context, walletID uuid.UUID, providerLinkingID uuid.UUID) (bool, error) {
```

Comment:
Could consider this business logic/rules and place on service rather then datastore

Reply:
I think generally we've kept all db queries in datastore, but on a related note I opened #1670 since I agree business logic should not be in datastore and noticed some reputation logic was.